perllib/FixMyStreet/DB/Result/Problem.pm

```diff
@@ -868,4 +868,20 @@ sub get_time_spent {
     return $admin_logs ? $admin_logs->get_column('sum_time_spent') : 0;
 }
 
+=head2 get_cobrand_logged
+
+Get a cobrand object for the cobrand the problem was logged for.
+
+e.g. if a problem was logged at www.fixmystreet.com, this will be a
+FixMyStreet::Cobrand::FixMyStreet object.
+
+=cut
+
+sub get_cobrand_logged {
+    my $self = shift;
+    my $cobrand_class = FixMyStreet::Cobrand->get_class_for_moniker( $self->cobrand );
+    return $cobrand_class->new;
```

Comment:
I think this would be simpler as a read-only lazy `has` attribute, see PhotoSet.pm.